codec/src/main/java/io/netty/handler/codec/compression/BrotliEncoder.java

```diff
@@ -47,19 +72,32 @@ public BrotliEncoder() {
     /**
      * Create a new {@link BrotliEncoder} Instance
      *
-     * @param parameters {@link Encoder.Parameters} Instance
+     * @param brotliOptions {@link BrotliOptions} to use.
      */
-    public BrotliEncoder(Encoder.Parameters parameters) {
-        this.parameters = ObjectUtil.checkNotNull(parameters, "Parameters");
+    public BrotliEncoder(BrotliOptions brotliOptions) {
+        this(brotliOptions.parameters());
     }
 
     /**
      * Create a new {@link BrotliEncoder} Instance
      *
-     * @param brotliOptions {@link BrotliOptions} to use.
+     * @param parameters {@link Encoder.Parameters} Instance
      */
-    public BrotliEncoder(BrotliOptions brotliOptions) {
-        this(brotliOptions.parameters());
+    public BrotliEncoder(Encoder.Parameters parameters) {
+        this.parameters = ObjectUtil.checkNotNull(parameters, "Parameters");
+    }
+
+    @Override
+    public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+        pluggableBuffer = new PluggableByteBufOutputStream();
+        brotliEncoderChannel = new BrotliEncoderChannel(Channels.newChannel(pluggableBuffer), parameters);
```

Comment:
Looks like this constructor wants a `WritableByteChannel`. Why not implement that, instead of another `OutputStream` for `ByteBuf`?